api.js

```diff
@@ -414,12 +415,21 @@ API.prototype.addCustomSpanAttribute = function addCustomSpanAttribute(key, valu
  *  The error to be traced.
  * @param {object} [customAttributes]
  *  Optional. Any custom attributes to be displayed in the New Relic UI.
- * @returns {false|undefined} Retruns false when disabled/errored, otherwise undefined
+ * @param {boolean} expected
+ *  Optional. False by default. True if the error is expected, meaning it should be collected
+ *  for error events and traces, but should not impact error rate.
+ * @returns {false|undefined} Returns false when disabled/errored, otherwise undefined
  */
-API.prototype.noticeError = function noticeError(error, customAttributes) {
+API.prototype.noticeError = function noticeError(error, customAttributes, expected = false) {
   const metric = this.agent.metrics.getOrCreateMetric(NAMES.SUPPORTABILITY.API + '/noticeError')
   metric.incrementCallCount()
 
+  // let users skip the custom attributes if they want
+  if (customAttributes && typeof customAttributes === 'boolean') {
```

Comment:
I waffled on making it optional, but I could envision a developer thinking primarily about noticing and expecting an error, but not having custom attributes to define. I do prefer taking one hash parameter, instead of positional. But yeah, that'd be a big breaking change for anyone using this API.

Reply:
To be clear, I totally agree with the reasoning for making it optional, just wish that we could have had this discussion sooner / planned accordingly for a breaking change. I don't think that discussing a breaking change should block this work.